include/ucl.h

```diff
@@ -469,8 +469,9 @@ UCL_EXTERN bool ucl_object_insert_key_merged (ucl_object_t *top, ucl_object_t *e
  * Reserve space in ucl array or object for `elt` elements
  * @param obj object to reserve
  * @param reserved size to reserve in an object
+ * @return 0 on success, -1 on failure (i.e. ENOMEM)
  */
-UCL_EXTERN void ucl_object_reserve (ucl_object_t *obj, size_t reserved);
+UCL_EXTERN int ucl_object_reserve (ucl_object_t *obj, size_t reserved);
```

Comment:
Why not bool as the rest of the API? I understand that for libc API the normal practice is to return `-1` on error. However, these interfaces created in the dark ages where there were normal `bool` type and it was possible to return both result and error marker in a single variable. Since we have no such a semantics here I'd suggest to use `bool` type here.